sctp/src/queue/payload_queue.rs

```diff
@@ -65,7 +75,7 @@ impl PayloadQueue {
     /// pop pops only if the oldest chunk's TSN matches the given TSN.
     pub(crate) fn pop(&mut self, tsn: u32) -> Option<ChunkPayloadData> {
         if Some(&tsn) == self.sorted.front() {
```

Comment:
you can pop_front here. that way you won't need to do it later and would be able to remove assert_eq! completely.

Reply:
I don't think so. Doing so broke a few tests, and it wouldn't do what the signature says anymore.
Note that this is not binding `tsn` as part of a pattern but that this actually an equality check, and pop is only called if the tsns match. That tripped me up the first time I changed this part.

I should just remove the assert_eq! it doesn't do anything.